modules/partition-replicator/src/main/java/org/apache/ignite/internal/partition/replicator/PartitionReplicaLifecycleManager.java

```diff
@@ -360,6 +445,415 @@ private CompletableFuture<List<Assignments>> getOrCreateAssignments(
+            ZonePartitionId zonePartitionId,
+            Set<Assignment> stableAssignments
+    ) {
+        // Update raft client peers and learners according to the actual assignments.
+        if (replicaMgr.isReplicaStarted(zonePartitionId)) {
+            replicaMgr.getReplica(zonePartitionId).join()
+                    .raftClient().updateConfiguration(fromAssignments(stableAssignments));
+        }
+
+        return nullCompletedFuture();
+    }
+
+    private CompletableFuture<Void> stopAndDestroyPartitionAndUpdateClients(
+            ZonePartitionId zonePartitionId,
+            Set<Assignment> stableAssignments,
+            Assignments pendingAssignments,
+            boolean isRecovery
+    ) {
+        CompletableFuture<Void> clientUpdateFuture = isRecovery
+                // Updating clients is not needed on recovery.
+                ? nullCompletedFuture()
+                : updatePartitionClients(zonePartitionId, stableAssignments);
+
+        boolean shouldStopLocalServices = (pendingAssignments.force()
+                ? pendingAssignments.nodes().stream()
+                : Stream.concat(stableAssignments.stream(), pendingAssignments.nodes().stream())
+        )
+                .noneMatch(assignment -> assignment.consistentId().equals(localNode().name()));
+
+        if (shouldStopLocalServices) {
+            return allOf(
+                    clientUpdateFuture,
+                    stopAndDestroyPartition(zonePartitionId)
+            );
+        } else {
+            return clientUpdateFuture;
+        }
+    }
+
+    private CompletableFuture<?> stopAndDestroyPartition(ReplicationGroupId zonePartitionId) {
+        return stopPartition(zonePartitionId);
+    }
+
+    private CompletableFuture<Void> handleChangePendingAssignmentEvent(
+            Entry pendingAssignmentsEntry,
+            long revision,
+            boolean isRecovery
+    ) {
+        if (pendingAssignmentsEntry.value() == null || pendingAssignmentsEntry.empty()) {
+            return nullCompletedFuture();
+        }
+
+        int partId = extractPartitionNumber(pendingAssignmentsEntry.key());
+        int zoneId = extractZoneId(pendingAssignmentsEntry.key(), PENDING_ASSIGNMENTS_PREFIX);
+
+        var zonePartitionId = new ZonePartitionId(zoneId, partId);
+
+        // Stable assignments from the meta store, which revision is bounded by the current pending event.
+        Assignments stableAssignments = stableAssignments(zonePartitionId, revision);
+
+        Assignments pendingAssignments = Assignments.fromBytes(pendingAssignmentsEntry.value());
+
+        if (!busyLock.enterBusy()) {
+            return CompletableFuture.<Void>failedFuture(new NodeStoppingException());
+        }
+
+        try {
+            if (LOG.isInfoEnabled()) {
+                var stringKey = new String(pendingAssignmentsEntry.key(), UTF_8);
+
+                LOG.info("Received update on pending assignments. Check if new raft group should be started [key={}, "
```

Comment:
Logs are a bit outdated, it's not only raft group, let's reformulate it to replication group. Not only here I guess.

Reply:
changed the message